y3prediction/prediction.py

```diff
@@ -1369,6 +1369,11 @@ def my_partitioner(mesh, tag_to_elements, num_ranks):
             wall_vol_discr, actx, wall_tag_to_elements["wall"])
         wall_surround_mask = None
 
+    # initialize the wall heat capacity
+    wall_cp = wall_insert_cp * wall_insert_mask
+    if wall_surround_mask is not None:
+        wall_cp = wall_cp + wall_surround_cp * wall_surround_mask
```

Comment:
Gotta be careful here, flash and shock don't have a surround.  That's why I massaged this bit.